src/wallet/wallet_tx_builder.cpp

```diff
@@ -0,0 +1,691 @@
+                    }
+                } else {
+                    resolutionError = AddressResolutionError::RevealingSaplingAmountNotAllowed;
+                }
+            },
+            [&](const UnifiedAddress& ua) {
+                if (canResolveOrchard
+                    && ua.GetOrchardReceiver().has_value()
+                    && (strategy.AllowRevealedAmounts() || payment.GetAmount() < maxOrchardAvailable)
+                    ) {
+                    resolvedPayments.emplace_back(
+                        ua, ua.GetOrchardReceiver().value(), payment.GetAmount(), payment.GetMemo(), false);
+                    if (!strategy.AllowRevealedAmounts()) {
+                        maxOrchardAvailable -= payment.GetAmount();
+                    }
+                    orchardOutputs += 1;
+                } else if (ua.GetSaplingReceiver().has_value()
+                    && (strategy.AllowRevealedAmounts() || payment.GetAmount() < maxSaplingAvailable)
+                    ) {
+                    resolvedPayments.emplace_back(
+                        ua, ua.GetSaplingReceiver().value(), payment.GetAmount(), payment.GetMemo(), false);
+                    if (!strategy.AllowRevealedAmounts()) {
+                        maxSaplingAvailable -= payment.GetAmount();
+                    }
+                } else {
+                    if (strategy.AllowRevealedRecipients()) {
+                        if (ua.GetP2SHReceiver().has_value()) {
+                            resolvedPayments.emplace_back(
+                                ua, ua.GetP2SHReceiver().value(), payment.GetAmount(), std::nullopt, false);
+                        } else if (ua.GetP2PKHReceiver().has_value()) {
+                            resolvedPayments.emplace_back(
+                                ua, ua.GetP2PKHReceiver().value(), payment.GetAmount(), std::nullopt, false);
+                        } else {
+                            // There are no receivers in this UA, which should be impossible.
+                            assert(false);
+                        }
+                    } else if (strategy.AllowRevealedAmounts()) {
+                        resolutionError = AddressResolutionError::TransparentReceiverNotAllowed;
+                    } else {
+                        resolutionError = AddressResolutionError::RevealingReceiverAmountsNotAllowed;
+                    }
+                }
+            }
+        }, payment.GetAddress());
+
+        if (resolutionError.has_value()) {
+            return resolutionError.value();
+        }
+    }
+    auto resolved = Payments(resolvedPayments);
+
+    if (spendableMut.HasTransparentCoinbase() && resolved.HasTransparentRecipient()) {
+        return AddressResolutionError::TransparentRecipientNotAllowed;
+    }
+
+    if (orchardOutputs > this->maxOrchardActions) {
+        return ExcessOrchardActionsError(spendableMut.orchardNoteMetadata.size(), this->maxOrchardActions);
```

Comment:
Looks like a copy-paste bug from the spend-side check below:
```suggestion
        return ExcessOrchardActionsError(orchardOutputs, this->maxOrchardActions);
```